sbnana/CAFAna/XSec/Flux.cxx

```diff
@@ -90,6 +93,70 @@ namespace ana
     return ret;
   }
 
+  //----------------------------------------------------------------------
+  void FluxTimesNuclei::SaveTo(TDirectory* dir, const std::string& name) const
+  {
+    TDirectory* tmp = gDirectory;
+
+    dir = dir->mkdir(name.c_str()); // switch to subdir
+    dir->cd();
+
+    TObjString("FluxTimesNuclei").Write("type");
+
+    TVectorD pdg(1);
+    pdg[0] = fPdg;
+    pdg.Write("pdg");
+
+    Spectrum::SaveTo(dir, "spectrum");
+
+    dir->Write();
+    delete dir;
+
+    tmp->cd();
+  }
+
+  //----------------------------------------------------------------------
+  std::unique_ptr<FluxTimesNuclei> FluxTimesNuclei::LoadFrom(TDirectory* topdir, const std::string& name)
+  {
+    std::unique_ptr<TDirectory> dir(topdir->GetDirectory(name.c_str())); // switch to subdir
```

Comment:
I think this `topdir` pattern is from somewhere that needed to inspect the contents of topdir. You can just call the parameter `dir` and switch what it points to like https://github.com/cafana/CAFAnaCore/blob/master/CAFAna/Core/Spectrum.cxx#L583